app/controllers/animals_controller.rb

```diff
@@ -0,0 +1,39 @@
+class AnimalsController < ApplicationController
+    def index
+        animals = Animal.all
+        render json: animals
+    end
+    def show
+        animal = Animal.find(params[:id])
+        render json: animal
+    end
+    def create
+        animal = Animal.create(animal_params)
+        if animal.valid?
+            render json: animal
+        else
+        render json: animal.errors
+        end
+    end
+    def update
+        animal = Animal.find(params[:id])
+        animal.update(animal_params)
+        if animal.valid?
+            render json: animal
+        else
+        render json: animal.errors
```

Comment:
indentation

Reply:
I guess you caught me copy and pasting :) with the same indentation